### PR TITLE
Add intersection observer for lazy loading of images

### PR DESCRIPTION
**Pull request by mondeja**
Contributes to #284

I'm not sure if this solution will avoid all loading problems, but seems better than the current implementation.  Added a smooth opacity transition on image loads too.

### CODE REVIEW

File: public/scripts/iobserver.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    // Browser does not support IntersectionObserver,
+    // so we load using the loading=lazy attribute
+    for (const img of images) {
+      const src = img.getAttribute('d-src');
+      img.removeAttribute('d-src');
+      img.onload = () => {
+        img.classList.add('ld');
+      };
+      img.setAttribute('loading', 'lazy');
+      img.setAttribute('src', src);
```
Comment (kidonng) [inline public/scripts/iobserver.js:42]: This seems unnecessary because `loading=lazy` support implies `IntersectionObserver` support.

Comment (mondeja) [reply public/scripts/iobserver.js:42]: Thanks, removing that on #293

Comment (mondeja) [reply public/scripts/iobserver.js:42]: This implementation is not exhaustive, if a browser like IE11 does not support `loading=lazy` just loads the images as normal ignoring the attribute. It is only added here in the case that IE11 or other browsers starts supporting it.

Comment (kidonng) [reply public/scripts/iobserver.js:42]: > It is only added here in the case that IE11 or other browsers starts supporting it.

[IE is deprecated](https://endoflife.date/internet-explorer), while other browsers all support it in their latest versions.

I'm just nit-picking that `'IntersectionObserver' in window` is practically always true nowadays.

**Pull request merged**